### PR TITLE
Fix/configs update

### PR DESCRIPTION
**Pull request by savingoyal**

### CODE REVIEW

File: metaflow/cli.py [hunk 1]
```diff
@@ -354,19 +382,76 @@ def start(
...
+    if (
+        hasattr(ctx, "saved_args")
+        and ctx.saved_args
+        and ctx.saved_args[0] == "resume"
+        and getattr(ctx.obj, "has_config_options", False)
+    ):
+        # In the case of resume, we actually need to load the configurations
+        # from the resumed run to process them. This can be slightly onerous so check
+        # if we need to in the first place
+        if getattr(ctx.obj, "has_cl_config_options", False):
```
Comment (savingoyal) [inline metaflow/cli.py:396]: what's cl?

Comment (romain-intel) [reply metaflow/cli.py:396]: command line. I can make it longer. Basically, this part of the code is only hit if the user explicitly specified a `--config` or `--config-value` option (as opposed to being derived from defaults).

File: metaflow/cli.py [hunk 2]
```diff
@@ -519,10 +591,13 @@ def _check(echo, graph, flow, environment, pylint=True, warnings=False, **kwargs
...
 def print_metaflow_exception(ex):
     echo_always(ex.headline, indent=True, nl=False, bold=True)
+    location = ""
+    if ex.source_file is not None:
+        location += " in file %s" % ex.source_file
     if ex.line_no is None:
-        echo_always(":")
-    else:
-        echo_always(" on line %d:" % ex.line_no, bold=True)
+        location += " on line %d" % ex.line_no
```
Comment (saikonen) [inline metaflow/cli.py:598]: This part fails as line_no is None but the string is expecting an integer.

Comment (saikonen) [reply metaflow/cli.py:598]: should be able to hit this with any pypi/conda flow and failing to apply `--environment conda` on the CLI at least.

**Review by saikonen — CHANGES_REQUESTED** [review]

File: metaflow/cli.py (hunk 2 above)
Comment (romain-intel) [reply metaflow/cli.py:598]: well yes, I am an idiot -- the condition is reversed above. Let me fix that.

**Review by saikonen — APPROVED** [review]

**Pull request merged**